### PR TITLE
post messages to office-hours if its open

### LINKED ISSUE #163: [Feature] If !ask is used while office-hours is open, post directly to office-hours
This would be a nice-to-have.

### PR DESCRIPTION
**Pull request by stelar7**
Fixes #163

### CODE REVIEW

File: src/OfficeHours.ts [hunk 1]
```diff
@@ -56,7 +57,7 @@ export default class OfficeHours {
         this.data = fileBackedObject(officeHoursData);
         console.log("Successfully question file.");
 
-        bot.on("ready", this.setupOpenState.bind(this));
+        bot.on("ready", this.setupOpenState.bind(this, bot));
```
Comment (molenzwiebel) [inline src/OfficeHours.ts]: Just use `this.bot`, no need to bind the argument.

File: src/OfficeHours.ts [hunk 2]
```diff
@@ -109,25 +110,26 @@ export default class OfficeHours {
         }
 
         message.delete();
-        this.close(message.channel);
+        this.close(this.channel);
     }
 
-    private setupOpenState() {
-        this.guild = this.bot.guilds.get(this.sharedSettings.server) as Discord.Guild;
+    private setupOpenState(bot: Discord.Client) {
```
Comment (molenzwiebel) [inline src/OfficeHours.ts]: Again, bot is already in `this.bot`.

File: src/OfficeHours.ts [hunk 3]
```diff
@@ -109,25 +110,26 @@ export default class OfficeHours {
...
+        this.close(this.channel);
     }
 
     private setupOpenState() {
-        this.guild = this.bot.guilds.get(this.sharedSettings.server) as Discord.Guild;
+
+        this.guild = this.bot.guilds.get(this.sharedSettings.server)!;
         if (!this.guild) {
-            console.error(`OfficeHours: Unable to find server with ID: ${this.sharedSettings.server}`);
+            console.error(`Office-Hours: Unable to find server with ID: ${this.sharedSettings.server}`);
```
Comment (Querijn) [inline src/OfficeHours.ts]: Why was this necessary

**Comment by Querijn** [comment]
What's with the other changes that are in this request? Are they necessary?

**Comment by stelar7** [comment]
I needed the office-hours channel in the function, so i made use of the new variable everywhere

File: src/OfficeHours.ts (hunk 3 above)
Comment (stelar7) [reply src/OfficeHours.ts]: it wasnt 🤔

**Comment by Querijn** [comment]
I'm merging it, but try to not too put too much unrelated stuff into a commit
